### ISSUE TITLE
Screenreader doesn't announce selected date when range is active

### ISSUE THREAD

**Issue by JoseeWouters** [Hacker0x01/react-datepicker#5125]
**Describe the bug**
We are using two date pickers to have the option to select a specific date range with a start date an end date. We have used several of the available Aria labels to make sure a screenreader announces correctly the selected date and also the focused date.
But if you have a date selected, for example september 1st to september 25th and after this you select september 15th, the screenreader will not announce this as selected. 

**To Reproduce**

1. Make sure to have a screenreader active (I'm using NVDA)
1. In a date picker where you can select a date range
2. select a start date
3. select an end date
4. The screenreader will announce the selected dates correctly
5. After this selection, select a new date in this date range
6. the screenreader will not announce this new date

**Expected behavior**
I expect the screenreader to announce the newly selected date, always, regardless of other previous selections

**Screenshots**

**Desktop (please complete the following information):**

- OS: Windows
- Browser [e.g. chrome, safari] Chrome
- Version 129

**Screenreader**
- NVDA
- Version 2023.2

**Additional context**

**Comment by ahaws-govcio** [comment]
👍 
On top of this, the screen reader does not announce the keyboard navigation of the dates, therefore, the users doesn't know what date is high-lighted by the keyboard navigation.
